pkg/sql/create_table.go

```diff
@@ -90,7 +91,16 @@ func (p *planner) getNonTemporarySchemaForCreate(
 	case catalog.SchemaPublic:
 		return sc, nil
 	case catalog.SchemaUserDefined:
-		return p.Descriptors().MutableByID(p.txn).Schema(ctx, sc.GetID())
+		sc, err := p.Descriptors().MutableByID(p.txn).Schema(ctx, sc.GetID())
+		if err != nil {
+			return nil, err
+		}
+		// Exit early with an error if the schema is undergoing a declarative schema
+		// change.
+		if catalog.HasConcurrentDeclarativeSchemaChange(sc) {
+			return nil, scerrors.ConcurrentSchemaChangeError(sc)
```

Comment:
could we make this error occur for _all_ legacy schema changes? (e.g., put this check in a common code path like writeSchemaChange.) seems like it might save us a lot of headaches. or would that be too much of a behavior change?

Reply:
writeSchemaChange already has this logic, so that covers most if not all modification operations. The problem in this case is that the name resolution logic for new objects does not need to modify the parent objects. So, there isn't a singular code path outside of shared logic for resolving schema / databases for parent objects.
<!-- Reviewable comment -OCo9gbVCCidHnF2sbXe:bocmlfo -->
<!-- Sent from Reviewable.io -->